### PR TITLE
[Play-193] Text Input Kit Typescript Conversion

### PR DESCRIPTION
**Pull request by sergiopvilar**
React Kit Coverage
#### Screens

Captura de Tela 2022-05-24 às 18 03 49

#### Breaking Changes

Possible breaking changes:

**Caption:**

 - dark: property doesn't exist and needed to be removed

**Body:**

 - variant: property required and added as  null

#### Runway Ticket URL

[https://nitro.powerhrg.com/runway/backlog_items/PLAY-193](https://nitro.powerhrg.com/runway/backlog_items/PLAY-193)

#### How to test this

No behavior change is expected but regresion tests might be required.

#### Checklist:

- [ ] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [ ] **SCREENSHOT** Please add a screen shot or two.
- [ ] **SPECS** Please cover your changes with specs.
- [ ] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_text_input/_text_input.tsx [hunk 1]
```diff
@@ -48,7 +47,7 @@ const TextInput = (props: TextInputProps, ref: React.ElementRef<"input">) => {
     inline = false,
     name,
     label,
-    onChange = () => {},
+    onChange = () => { void 0 },
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_text_input/_text_input.tsx:50]: Why was this necessary?

File: playbook/app/pb_kits/playbook/pb_text_input/_text_input.tsx [hunk 2]
```diff
@@ -104,34 +103,38 @@ const TextInput = (props: TextInputProps, ref: React.ElementRef<"input">) => {
     <React.Fragment>
       <Flex
           className={`add-on-${addOnAlignment} ${borderCss}`}
-          inline="flex-container"
+          inline
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_text_input/_text_input.tsx:106]: Wondering why this had to be removed here.

Comment (sergiopvilar) [reply playbook/app/pb_kits/playbook/pb_text_input/_text_input.tsx:106]: The `inline` prop on Flex is a boolean so I wasn't able to make Typescript compile without changing this. I assumed that everything different from false is being understood as `true` but I wasn't table to put `true` doe to this rule:

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md

File: playbook/app/pb_kits/playbook/pb_text_input/_text_input.tsx [hunk 3]
```diff
@@ -140,21 +143,17 @@ const TextInput = (props: TextInputProps, ref: React.ElementRef<"input">) => {
...
-              status="negative"
-              text={error}
-          />
-        </If>
+        {render}
+
+        {error && <Body
+            status="negative"
+            text={error}
+            variant={null}
```
[the remark this answers is not in the document]
Comment (sergiopvilar) [reply playbook/app/pb_kits/playbook/pb_text_input/_text_input.tsx:154]: As long as the default value is null I think is fine. Not specifying that will turn the value into `undefined`.

**Review by thestephenmarshall — APPROVED** [review]
